app/scripts/lib/createMetaRPCHandler.js

```diff
@@ -3,7 +3,11 @@ import { isManifestV3 } from '../../../shared/modules/mv3.utils';
 
 const createMetaRPCHandler = (api, outStream, store, localStoreApiWrapper) => {
   return async (data) => {
-    if (outStream._writableState.ended) {
+    if (
+      !outStream.writable ||
+      outStream.destroyed ||
+      outStream._writableState.ended
+    ) {
```

Comment:
1. Do you have any docs supporting this pattern? I can't find a reference to "destroyed" in the [relevant Node docs](https://nodejs.org/dist/v10.18.1/docs/api/stream.html).
2. Do we need to continue relying on `_writableState.ended`?
3. This looks like a candidate for a smol utility function, somewhere in `shared/` or maybe even `@metamask/utils`.

Reply:
1 & 2: Conservative backwards-compat for legacy streams until we are confident that those never come into play here (appearing in v3 implementation as well).

3: Sure, `app/scripts/lib/stream-utils.js` sounds appropriate to me?

Reply:
Broke out as `isStreamWritable`.